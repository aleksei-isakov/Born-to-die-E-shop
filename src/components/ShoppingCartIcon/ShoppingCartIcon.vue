<template>
  <div class="cart__wrapper">
    <BaseButtonRouter class="link" path="/cart">
      <BaseCustomIcon :width="width" icon="shopping_basket_white_24dp" />

      <span
        class="cart__is__empty"
        :class="{ cart__items__quantity: cartItemsCount.length !== 0 }"
      >
        {{ cartItemsCount.length }}
      </span>
    </BaseButtonRouter>
  </div>
</template>

<script>
import { BaseCustomIcon, BaseButtonRouter } from '@/base_components/';

export default {
  name: 'ShoppingCartIcon',

  components: {
    BaseButtonRouter,
    BaseCustomIcon
  },

  props: {
    width: {
      type: [String, Number],
      default: '55'
    }
  },

  computed: {
    cartItemsCount() {
      return [];
    }
  }
};
</script>

<style lang="scss" scoped>
.link {
  display: flex;
  position: relative;
}

.cart__is__empty {
  display: none;
}

.cart__items__quantity {
  position: absolute;
  top: 17px;
  left: 17px;
  border-radius: 50%;
  background-color: #f30808;
  min-width: 17px;
  height: 17px;
  display: inline-block;
  color: #fcfcfd;
  font-weight: bold;
  font-size: 12px;
}
</style>
