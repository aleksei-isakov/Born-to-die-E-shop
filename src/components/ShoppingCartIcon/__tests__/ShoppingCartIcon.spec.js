import { createLocalVue, shallowMount } from '@vue/test-utils';
import ShoppingCartIcon from '@/components/ShoppingCartIcon/ShoppingCartIcon';
import Vuex from 'vuex';
import ShoppingCartModule from '@/store/modules/shoppingcart';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('ShoppingCartIcon', () => {
  let store;
  let wrapper;

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        ShoppingCartModule: {
          namespaced: true,
          state: ShoppingCartModule.state,
          getters: ShoppingCartModule.getters,
          actions: ShoppingCartModule.actions
        }
      }
    });
  });
  test('should import BaseCustomIcon', () => {
    wrapper = shallowMount(ShoppingCartIcon, {
      localVue,
      store
    });
    expect(wrapper.find('BaseCustomIcon')).toBeTruthy();
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
