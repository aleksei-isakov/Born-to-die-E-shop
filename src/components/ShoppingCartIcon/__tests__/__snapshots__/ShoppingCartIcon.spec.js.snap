// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCartIcon should match snapshot 1`] = `undefined`;
